test/test.c

```diff
@@ -15868,7 +15868,202 @@ test_SSLVerify(void)
 
     nats_Sleep(100);
 
-    serverPid = _startServer("nats://127.0.0.1:4443", "-config tls.conf", true);
+    serverPid = _startServer("nats://127.0.0.1:4443", "-config tlsverify.conf", true);
+    CHECK_SERVER_STARTED(serverPid);
+
+    natsMutex_Lock(args.m);
+    while ((s == NATS_OK) && !(args.reconnected))
+        s = natsCondition_TimedWait(args.c, args.m, 2000);
+    natsMutex_Unlock(args.m);
+
+    if (s == NATS_OK)
+        s = natsConnection_PublishString(nc, "foo", "test");
+    if (s == NATS_OK)
+        s = natsConnection_Flush(nc);
+    testCond(s == NATS_OK)
+
+    natsConnection_Destroy(nc);
+    natsOptions_Destroy(opts);
+
+    _destroyDefaultThreadArgs(&args);
+
+    _stopServer(serverPid);
+#else
+    test("Skipped when built with no SSL support: ");
+    testCond(true);
+#endif
+}
+
+static void
+test_SSLLoadCAFromMemory(void)
+{
+#if defined(NATS_HAS_TLS)
+    natsStatus          s;
+    natsConnection      *nc       = NULL;
+    natsOptions         *opts     = NULL;
+    natsPid             serverPid = NATS_INVALID_PID;
+    natsBuffer          *certBuf  = NULL;
+    struct threadArg    args;
+
+    s = nats_ReadFile(&certBuf, 10000, "certs/ca.pem");
+    if (s == NATS_OK)
+        s = _createDefaultThreadArgsForCbTests(&args);
+    if (s == NATS_OK)
+        opts = _createReconnectOptions();
+    if (opts == NULL)
+        FAIL("Unable to create reconnect options!");
+
+    test("Check NULL certs: ");
+    s = natsOptions_SetCATrustedCertificates(opts, NULL);
+    testCond(s == NATS_INVALID_ARG);
+
+    test("Check empty certs: ");
+    s = natsOptions_SetCATrustedCertificates(opts, "");
+    testCond(s == NATS_INVALID_ARG);
+
+    test("Check invalid cert: ");
+    s = natsOptions_SetCATrustedCertificates(opts, "invalid");
+    testCond(s == NATS_SSL_ERROR);
+
+    serverPid = _startServer("nats://127.0.0.1:4443", "-config tlsverify.conf", true);
+    CHECK_SERVER_STARTED(serverPid);
+
+    test("Check that connect succeeds with proper certs: ");
+    s = natsOptions_SetReconnectedCB(opts, _reconnectedCb, &args);
+    if (s == NATS_OK)
+        s = natsOptions_SetURL(opts, "nats://localhost:4443");
+    if (s == NATS_OK)
+        s = natsOptions_SetSecure(opts, true);
+    // For test purposes, we provide the CA trusted certs
+    if (s == NATS_OK)
+    {
+        s = natsOptions_SetCATrustedCertificates(opts, (const char*) natsBuf_Data(certBuf));
+        // Demonstrate that we can free the memory after this call..
```

Comment:
This comment was very helpful to me, i was about to comment and then realized i didn't have to and am anyway to say I like the comment